secrets/secrets.go

```diff
@@ -298,3 +321,35 @@ func NewSecrets(r io.Reader) (*Secrets, error) {
 	}
 	return secrets, nil
 }
+
+func csiPathParser(dir fs.FS) (Document, error) {
+	secretsDocument := Document{
+		Secrets: make(map[string]GenericSecret),
+	}
+	err := fs.WalkDir(
+		dir,
+		".",
+		func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if d.IsDir() {
+				return nil
+			}
+			file, err := dir.Open(path)
+			if err != nil {
+				return err
+			}
+			defer file.Close()
+
+			var secretFile CSIFile
+			err = json.NewDecoder(file).Decode(&secretFile)
+			if err != nil {
+				return err
```

Comment:
🔕 bare returns like this can make errors difficult to trace back if they don't contain sufficient context. I'm not going to audit everything in here, but something to keep in mind for the future. Instead:

a) if the error message contains sufficient context, add a comment to let code reviewers and future readers know that you thought about this:

```go
return err // contains context
```

or, b) wrap the error with a helpful message:

```go
return fmt.Errorf("decoding %q: %s", path, err)
```

Reply:
agreed but we should use `%w` for `err`.